client/src/pages/club/match/EditMatch.tsx

```diff
@@ -1,9 +1,365 @@
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import KakaoMapSearch, { PlaceType } from '../../../components/kakao/KakaoMapSearch';
+import KakaoMapView from '../../../components/kakao/KakaoMapView';
+import S_Container from '../../../components/UI/S_Container';
+import { S_Input } from '../../../components/UI/S_Input';
+import {
+  S_Button,
+  S_ButtonGray,
+  S_EditButton,
+  S_SelectButton,
+  S_NegativeButton
+} from '../../../components/UI/S_Button';
+import { S_Description, S_Label, S_Text, S_Title } from '../../../components/UI/S_Text';
+import { S_NameTag } from '../../../components/UI/S_Tag';
+// import { StyledSelect } from '../../../components/UI/S_Select';
+import AddMemberPopUp from '../../../components/match/AddMemberPopUp';
+
+import { useForm } from "react-hook-form";
+import { MatchDatas, Record, TeamList } from './CreateMatch';
+
+const S_MapBackdrop = styled.div`
+  background-color: rgba(0, 0, 0, 0.3);
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 100;
+`;
+
+const S_MapView = styled.div`
+  display: flex;
+  flex-direction: column;
+  background-color: white;
+  width: 300px;
+  height: 320px;
+  border-radius: 20px;
+  padding: 20px;
+  section {
+    width: 260px;
+    border: 1px solid black;
+    border-radius: 20px;
+    padding: 10px;
+  }
+`;
+
+
 function EditMatch() {
-    return (
+  const {
+    register,
+    unregister,
+    // formState: { errors },
+    getValues,
+} = useForm({ mode: "onChange" });
+
+  const [matchDatas, setMatchDatas] = useState<MatchDatas>();
+
+  const [date, setDate] = useState<string>(new Date().toISOString().slice(0,10));
+  const [time, setTime] = useState<string>();
+  const [placeValue, setPlaceValue] = useState<PlaceType>();
+  //참가를 누른 멤버들
+  const candidates: string[] = ['박대운', '우제훈', '김은택', '김아애', '문채리', '전규언전규언전규','박대운2', '우제훈2', '김은택2', '김아애2', '문채리2', '전규언전'];
+  
+  //팀구성에 필요한 후보들(팀에 들어가거나 빠질 때 실시간 반영되는 리스트)
+  const [candidateList, setCandidateList] = useState(candidates);
+
+  const [teamList, setTeamList] = useState<TeamList[]>([{ id: 0, members: [] }]);
+  const [records, setRecords] = useState<Record[]>([]);
+
+  const [isOpenMapSetting, setIsOpenMapSetting] = useState(false);
+  const [isOpenMapView, setIsOpenMapView] = useState(false);
+
+  const [isOpenAddMember, setIsOpenAddMember] = useState(false);
+  const [addButtonIndex, setAddButtonIndex] = useState(0);
+  const [addButtonPos, setAddButtonPos] = useState([0,0]);
```

Comment:
이건 배열보다는 {x:0, y:0} 이 가독성이 더 좋을 것 같아요. 아래쪽 코드를 확인하다 보면 index 0, 1이 각각 어떤 값인지 헷갈릴 수 있을 것 같아요 ㅎㅎ
```suggestion
  const [addButtonPos, setAddButtonPos] = useState({x:0, y:0});
```

Reply:
참고하여 반영하겠습니다. 감사합니다!